src/VSCodeExtension/src/tests/unit/formatter.test.ts

```diff
@@ -0,0 +1,107 @@
+import "mocha";
+import * as assert from "assert";
+import { formatter } from "../../formatter/formatter";
+import { spaceAfterIf } from "../../formatter/rules/space-after-if";
+
+describe("formatter core", () => {
+  it("with no rules, the code is unchanged", () => {
+    const code = "using(qs=Qubit[3]){H(qs[0]);}";
+    const expectedCode = "using(qs=Qubit[3]){H(qs[0]);}";
+
+    assert.equal(formatter(code, []), expectedCode);
+  });
+
+  describe("space after if rule", () => {
+    it("adds space after if statement", () => {
+      const code = "if(1 == 1){H(qs[0]);}";
+      const expectedCode = "if (1 == 1){H(qs[0]);}";
+
+      assert.equal(formatter(code, [spaceAfterIf]), expectedCode);
+    });

```

Comment:
Nit:
```suggestion
    const rules = [spaceAfterIf];
    it("adds space after if statement", () => {
      const code = "if(1 == 1){H(qs[0]);}";
      const expectedCode = "if (1 == 1){H(qs[0]);}";

      assert.equal(formatter(code, rules), expectedCode);
    });
```

And re-use `rules` throughout test.

Reply:
I don't think that would make any noticieable difference, I think it is better to have the rule explicitly set, i.e. reading [spaceAfterIf] gives more information than rules. in the latter, you would potentially need to go and check for the definition for a simple concept?